### PR TITLE
adds mantle: tailwind variant to apply selectors in the lower specificity mantle css layer

### PR DESCRIPTION
**Pull request by cody-dot-js**

### CODE REVIEW

File: assets/mantle.css [hunk 1]
```diff
@@ -4,6 +4,13 @@
 @tailwind components;
 @tailwind utilities;
 
+/**
+ * This layer is lower specificity than non-layered styles, but higher
+ * specificity than any `:where()` styles. All tailwind classes will override
+ * the `mantle` layer since they are not in a layer.
+ */
+@layer mantle;
```
Comment (OfTheDelmer) [inline assets/mantle.css:12]: Great solution.

File: packages/tailwind-preset/src/tailwind-plugin-component-variant.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * will override the `mantle` layer since they are not in a layer.
+ */
+const mantleVariantPlugin = plugin(
+	// @ts-expect-error PluginAPI is not typed correctly, missing postcss
+	(api: PluginAPI & { postcss: Postcss }) => {
+		// @ts-expect-error addVariant is not typed correctly
+		api.addVariant("mantle", ({ container, separator }: { container: Root; separator: string }) => {
+			const mantleRule = api.postcss.atRule({
+				name: "layer",
+				params: "peepeepoopoo",
```
Comment (OfTheDelmer) [inline packages/tailwind-preset/src/tailwind-plugin-component-variant.ts:18]: Hahaha

**Review by OfTheDelmer — APPROVED** [review]

File: packages/tailwind-preset/src/tailwind-plugin-component-variant.ts (hunk 2 above)
Comment (cody-dot-js) [reply packages/tailwind-preset/src/tailwind-plugin-component-variant.ts:18]: lol forgot to change this back before pushing 
You found one of my debug strings

**Review by aaronshekey — APPROVED** [review]
🫡 Sure!

**Comment by cody-dot-js** [comment]
closing for now
we can revisit later if we want this or not

i think the better path forward is to build the mantle.css ahead of time

**Pull request closed without merging**